### PR TITLE
JBIDE-21023 ctrl+shft+G sometimes give NPE in cdi tooling

### PR DESCRIPTION
**Pull request by dazarov**

### CODE REVIEW

File: cdi/plugins/org.jboss.tools.cdi.ui/src/org/jboss/tools/cdi/ui/search/InjectionPointQueryParticipant.java [hunk 1]
```diff
@@ -66,7 +66,10 @@ private static boolean containsInSearchScope(QuerySpecification querySpecificati
 	}
 	
 	public static boolean containsInSearchScope(QuerySpecification querySpecification, ICDIElement element){
-		return containsInSearchScope(querySpecification, element.getResource().getProject().getFullPath());
+		if(element.getResource() != null){
```
Comment (fbricon) [inline cdi/plugins/org.jboss.tools.cdi.ui/src/org/jboss/tools/cdi/ui/search/InjectionPointQueryParticipant.java]: is it possible the resource has a null project?

Comment (dazarov) [inline cdi/plugins/org.jboss.tools.cdi.ui/src/org/jboss/tools/cdi/ui/search/InjectionPointQueryParticipant.java]: If resource is a workspace root. But it is not our case

**Comment by alexeykazakov** [comment]
Please:
1. Add a comment to the ICDIElement.getResource() that it may return null
2. Add steps to reproduce to the jira
3. Add automated tests
4. Check if we have other places where we don't check if ICDIElement.getResource() == null (if any)

**Comment by alexeykazakov** [comment]
Pushed to master and 4.3.x

**Pull request merged**